### PR TITLE
Add people breakdown on retention

### LINKED ISSUE #1191: Surface People on Retention
**Is your feature request related to a problem? Please describe.**
I want to be able to see who's being retained

**Describe the solution you'd like**
When I click on a square in the retention table, I want to see the people who have been retained

**Describe alternatives you've considered**
List below

**Additional context**

***Thank*** you for your feature request - we love each and every one!

### LINKED ISSUE #1205: Retention table improvements
### Is your feature request related to a problem? Please describe

I’d love to be able to (1) choose which event counts as activity and (2) choose the relevant period (eg. we care about weekly retention more than daily retention).

I've also noticed that the retention table doesn't actually load for me on app and crashes when it times out.

### Describe the solution you'd like

1. Allow users to choose the event that counts as retention
2. Allow filtering of both the interval and the period
3. Make sure the query is optimised

### Describe alternatives you've considered

-

### Additional context

cc @EDsCODE as you're looking at retention table

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes
- adds ability to click and breakdown who is retaining

Todo:
- [ ] order by people who go through flow
- [ ] paginate
- [ ] add people tests

## Checklist

- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)
- [ ] Cypress E2E tests (if applicable)

### CODE REVIEW

**Review by timgl — COMMENTED** [review]
Mostly looks good, however the error state really confused me here:
image

**Comment by EDsCODE** [comment]
I added a message when there are no users. Also, in the date filtering PR, I changed the percentage to show 0 when there's no one rather than 100 so it will be less confusing

Screen Shot 2020-07-17 at 10 22 44 AM

File: frontend/src/scenes/retention/RetentionTable.js [hunk 1]
```diff
@@ -46,7 +54,107 @@ export function RetentionTable({ logic }) {
...
+                        top: 20,
+                        minWidth: retention?.data[selectedRow]?.values[0]?.count === 0 ? '10%' : '90%',
+                        fontSize: 16,
+                    }}
+                    title={retention.data[selectedRow].date}
+                >
+                    {retention && !peopleLoading ? (
+                        <div>
+                            {retention?.data[selectedRow]?.values[0]?.count === 0 ? (
+                                <span>No Users during this period</span>
```
Comment (timgl) [inline frontend/src/scenes/retention/RetentionTable.js:83]: Nit but we don't capitalise users anywhere else I think

**Review by timgl — APPROVED** [review]

File: frontend/src/scenes/retention/retentionTableLogic.js [hunk 2]
```diff
@@ -24,6 +46,21 @@ export const retentionTableLogic = kea({
...
+        },
+        retention: {
+            updateRetention: (_, { retention }) => retention,
+        },
+        loadingMore: [
+            false,
+            {
+                updateLoadingMore: (state, { status }) => status,
+            },
+        ],
```
Comment (Twixes) [inline frontend/src/scenes/retention/retentionTableLogic.js:63]: Isn't this pattern (with `actions.updateLoadingMore`) kind of a Kea no-no? https://kea.js.org/docs/guide/concepts#listeners

Comment (EDsCODE) [reply frontend/src/scenes/retention/retentionTableLogic.js:63]: yup good point. it's unnecessary

**Review by Twixes — APPROVED** [review]

**Pull request merged**